### ISSUE TITLE
Create an experiment for the download funnel

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-frontend#10600]
Depends on https://github.com/mozilla/bedrock/issues/10207

Implement an experiment to test removing a step from the download funnel for non-Firefox users. Users in the test cohort would be served a link which points to a special page on mozilla.org which starts the download automatically, while those on the control cohort would be served a link which points to the /new page.

Because the special page, which will be a version of the /thanks page, is unable to handle users with IE 9 and below, we need to be able to exclude those users from the experiment. Ideally we would exclude them entirely, meaning they wouldn't be placed into either cohort and wouldn't affect the data we are collecting, but considering the small number of these users, it might be acceptable to simply force them into the control cohort. According to GA, we have had 40 users with IE 9 or below over the past 2 weeks.

**Comment by ioanarusiczki** [comment]
@bobsilverberg 

I tested on Edge/Chrome and so far here are my results with AMO dev on Win10:

- I see a new cookie set on AMO dev : `%7B%2220210531_download_funnel_experiment%22%3A%22new-link%22%7D`
- when the new link is set there's a source=amo parameter added

![query string parameters](https://user-images.githubusercontent.com/33448286/121901857-a1d61c80-ccdb-11eb-85af-5ff131ebb3b0.png)
- on download URL https://www.mozilla.org/firefox/download/thanks/, there's a blocked:csp error I could reproduce with Edge and Chrome

this is the network:
![the csp content error on edge](https://user-images.githubusercontent.com/33448286/121902090-d6e26f00-ccdb-11eb-9243-6ca8a258d259.png)
and the console:
![the error in console with edge](https://user-images.githubusercontent.com/33448286/121902102-db0e8c80-ccdb-11eb-822b-c90e675663d1.png)

- I tested and reproduced with VM - Sandbox this: when I have the new-link experiment set and click on a download button from AMO for the first time, I land on https://www.mozilla.org/en-US/firefox/download/new/.  I go back and click again a download button from AMO and now I am redirected correctly to https://www.mozilla.org/en-US/firefox/download/thanks/

![the new page not available1](https://user-images.githubusercontent.com/33448286/121902945-a64f0500-ccdc-11eb-9830-b5f887daa20a.gif)

- I did not test this on Android yet, but it should be available there too.
- the download is triggered - installs can be made, I'll test more around the way they're triggered tomorrow.

**Comment by bobsilverberg** [comment]
> @bobsilverberg
> 
> I tested on Edge/Chrome and so far here are my results with AMO dev on Win10:
> 
> * I see a new cookie set on AMO dev : `%7B%2220210531_download_funnel_experiment%22%3A%22new-link%22%7D`
> 
> * when the new link is set there's a source=amo parameter added
> 

Those are good.

> 
> * on download URL [mozilla.org/firefox/download/thanks](https://www.mozilla.org/firefox/download/thanks/), there's a blocked:csp error I could reproduce with Edge and Chrome
>

This is a CSP error you are seeing *on* mozilla.org, correct (i.e. not on AMO)? If that is the case this is something I guess we'd need to take up with them, but note that the page we are planning to direct to hasn't been fully implemented for us yet, so perhaps this will go away when it has been.

> 
> * I tested and reproduced with VM - Sandbox this: when I have the new-link experiment set and click on a download button from AMO for the first time, I land on [mozilla.org/en-US/firefox/download/new](https://www.mozilla.org/en-US/firefox/download/new/).  I go back and click again a download button from AMO and now I am redirected correctly to [mozilla.org/en-US/firefox/download/thanks](https://www.mozilla.org/en-US/firefox/download/thanks/)
>

This sounds like a bug. I have done some more work on the experiment infrastructure in https://github.com/mozilla/addons-frontend/pull/10665, during which I noticed something similar, so perhaps this will be fixed via that PR. Can you open a separate issue about this and we can track it there?
 
> 
> * I did not test this on Android yet, but it should be available there too.
> 
> * the download is triggered - installs can be made, I'll test more around the way they're triggered tomorrow.

As discussed in the "For QA:" section above, the code for this new `/thanks/` page has not been completed yet, so while it probably will work for downloading Firefox, there will be other issues with is (such as not including the attribution code). I think for now all we need verified is that the experiment works and does direct users to different URLs based on which cohort they are in, and that we don't necessarily need to verify the download triggering part on mozilla.org until they have completed their work to support our use case.

**Comment by ioanarusiczki** [comment]
@bobsilverberg Ok, I'm closing this issue because I see the experiment present on Android too. For now that's enough. 

Verified on AMO dev - Android 10.0, Win10

**Comment by bobsilverberg** [comment]
https://github.com/mozilla/bedrock/issues/10207 has landed, which means we should be able to test the new link. I have removed the "verified fixed" label and changed the milestone to 2021.07.15, which I think is all we need to do to trigger this being re-verified by QA. It would be great to test this on dev at this point, making sure that attribution works correctly, which should include a test for an RTAMO add-on, and then we can check telemetry to see whether our experiment and variation information made it through as well.  If we are satisfied that it works on dev, we can enable it on stage and prod for the next push.

cc @jvillalobos as an FYI

**Comment by ioanarusiczki** [comment]
@bobsilverberg  I've tested today the RTAMO part - it does not seem to be affected. I've been using Sandbox VM - I tested the flow with a recommended, by firefox and verified addons on AMO dev (FF91.0a1).

**Comment by ioanarusiczki** [comment]
@bobsilverberg
 
I verified this on non-FF browsers:

1. On Desktop - the behavior for the new-link is that once I am redirected on this page https://www.mozilla.org/en-US/firefox/download/thanks/ the Firefox .exe file is automatically downloaded on PC. If I choose to open the file the installation is triggered:

![the download](https://user-images.githubusercontent.com/33448286/125300305-c0313700-e2de-11eb-8675-42962fa86ce8.gif)

The current-link didn't change the behavior meaning that user is redirected to https://www.mozilla.org/en-US/firefox/new/ and to get the FF file "Download Firefox" must be clicked and then it goes to https://www.mozilla.org/en-US/firefox/download/thanks/

2. On Android - the new-link will redirect for a second to https://www.mozilla.org/en-US/firefox/download/thanks/?experiment=20210531_amo_download_funnel_experiment&utm_campaign=non-fx-button&utm_content=banner-download-button&utm_medium=referral&utm_source=addons.mozilla.org&utm_term=amo-fx-cta-new-link&variation=new-link&xv=amo and then it opens the Playstore 

![download ff android](https://user-images.githubusercontent.com/33448286/125301248-a8a67e00-e2df-11eb-9fe8-180548635a92.png)

The current-link goes to https://www.mozilla.org/en-US/firefox/new/ and the user must tap manually on the link to access the app store. 

This looks like the expected behavior on both platforms (in case it's not, please let me know).


2. Not related to AMO pages,  I'll mention that I still see a csp error in the network on the download page https://www.mozilla.org/en-US/firefox/download/thanks. I added screenshots [above](https://github.com/mozilla/addons-frontend/issues/10600#issuecomment-860706008).

3. About the telemetry side - if I can check something related to it, please let me know.

**Comment by bobsilverberg** [comment]
> @bobsilverberg I've tested today the RTAMO part - it does not seem to be affected. I've been using Sandbox VM - I tested the flow with a recommended, by firefox and verified addon on AMO dev (FF91.0a1).

I just want to confirm, you tested the RTAMO flow with the `/thanks/` link, correct? That's the one that could possibly be an issue.

**Comment by bobsilverberg** [comment]
> This looks like the expected behavior on both platforms (in case it's not, please let me know).
> 
> 1. Not related to AMO pages,  I'll mention that I still see a csp error in the network on the download page [mozilla.org/en-US/firefox/download/thanks](https://www.mozilla.org/en-US/firefox/download/thanks). I added screenshots [above](https://github.com/mozilla/addons-frontend/issues/10600#issuecomment-860706008).
>


I tried to reproduce the CSP error and was unable to do so on Firefox or Chrome (I don't have Edge). Perhaps this is because I am on a Mac?
 
> 2. About the telemetry side - if I can check something related to it, please let me know.

If you are able to check the telemetry ping that is sent after you install Firefox via the new `/thanks/` flow, then that would be good. The ping should include all of the usual AMO attribution data, plus `experiment` and `variation`.

**Comment by ioanarusiczki** [comment]
@bobsilverberg 
1.  I don't remember the experiment variant on my RTAMO tests but I repeated it now only with new-link.  after I install the release version I close it and install FF Nightly because it only works like that with -dev or -stage (I also add some prefs, the flow requires some workarounds). 
Result: the addons are visible and can be installed from about:welcome page

![RTAMO flow](https://user-images.githubusercontent.com/33448286/125451025-0e42557e-ec82-46d6-80d8-cde4f33310c9.gif)

2. About the page https://www.mozilla.org/en-US/firefox/download/thanks and the red csp message:

- on Edge: is reproducible every time the page is loaded into a tab, reloaded 

- on Chome: I saw it only once at the first load into a tab after I installed Chrome into VM Sandbox - on my PC right now it is not reproducible
- on Firefox: first time after I install Firefox on Sandbox I have a GA error, after reloading the page or into other tabs - the error is no longer there. 

![on Firefox](https://user-images.githubusercontent.com/33448286/125451950-ecd21c23-7a2d-41f7-8b4d-45d8a744ae9b.png)

**Note:** this page at refresh will download the installer.exe file again and again (probably expected). Also, the network error mentioned above don't seem to impact it's functionality.

3. After installing Firefox I went to about:telemetry -> environment data ->settings  and I see the attribution data present:
- click from the banner data:

![telemetry ping](https://user-images.githubusercontent.com/33448286/125452685-06cf0713-44c4-4864-a574-4f7c779d0cdc.png)

- click from an add-on detail page:

![telemetry ping2](https://user-images.githubusercontent.com/33448286/125452727-744488ce-6722-4940-8089-c72303ad6268.png)

 attribution.source is mozilla.org (probably it's fine, I started from dev-allizom.org) and I remember source being replaced with `xv`=amo as parameter but maybe these are 2 different things. Let me know if that's alright.

**Comment by bobsilverberg** [comment]
That all looks good, thanks @ioanarusiczki!
